esmvaltool/cmorizers/obs/cmorize_obs_woa.py

```diff
@@ -40,31 +41,84 @@ def _fix_data(cube, var):
-            cube *= 44.661 / 1000.  # Convert from ml/l to mol/m^3
+        if var in ['o2', 'po4', 'si', 'no3']:
+            cube /= 1000.  # Convert from umol/kg to mol/m^3
     return cube
 
 
-def extract_variable(var_info, raw_info, out_dir, attrs, year):
-    """Extract to all vars."""
-    var = var_info.short_name
-    cubes = iris.load(raw_info['file'])
-    rawvar = raw_info['name']
-
-    for cube in cubes:
-        if cube.var_name == rawvar:
-            fix_var_metadata(cube, var_info)
-            convert_timeunits(cube, year)
-            fix_coords(cube)
-            _fix_data(cube, var)
-            set_global_atts(cube, attrs)
-            save_variable(
-                cube, var, out_dir, attrs, unlimited_dimensions=['time'])
+def collect_files(in_dir, var, cfg):
+    """Compose input file list and download if missing."""
+    file_list = []
+    var_dict = cfg['variables'][var]
+    in_dir = os.path.join(in_dir, var_dict['name'])
+    # create list of monthly data
+    for mon in range(1, 13):
+        fname = var_dict['file'] + "{:02d}".format(mon) + '_01.nc'
+        in_file = os.path.join(in_dir, fname)
+
+        # download if missing
+        if not os.path.isfile(in_file):
+            if not os.path.isdir(in_dir):
+                os.makedirs(in_dir)
+            logger.info('Input file %s is missing. Start download ... ', fname)
+            url = os.path.join(cfg['attributes']['source'], var_dict['name'],
+                               'netcdf', var_dict['file'].split('_')[1],
+                               cfg['custom']['resolution'], fname)
+            url_file = requests.get(url)
+            open(in_file, 'wb').write(url_file.content)
+
+        file_list.append(in_file)
+
+    return file_list
+
+
+def extract_variable(in_files, out_dir, attrs, raw_info, cmor_table):
+    """Extract variables and create OBS dataset."""
+    var = raw_info['var']
+    var_info = cmor_table.get_variable(raw_info['mip'], var)
+    rawvar = raw_info['raw_var']
+    with catch_warnings():
+        filterwarnings(
+            action='ignore',
+            message='Ignoring netCDF variable .* invalid units .*',
+            category=UserWarning,
+            module='iris',
+        )
+        cubes = iris.load(in_files, rawvar)
+    iris.util.equalise_attributes(cubes)
+    cube = cubes.concatenate_cube()
+
+    # set reference time
+    year = raw_info['reference_year']
+    cube.coord('time').points = np.arange(0.5, 12.5)
+    cube.coord('time').units = Unit('months since ' + str(year) +
+                                    '-01-01 00:00:00',
+                                    calendar='gregorian')
+
+    fix_var_metadata(cube, var_info)
+    fix_coords(cube)
+    _fix_data(cube, var)
+    set_global_atts(cube, attrs)
+    save_variable(cube, var, out_dir, attrs, unlimited_dimensions=['time'])
+
+    # compute derived vars
+    if 'srf_var' in raw_info:
+        var_info = cmor_table.get_variable(raw_info['mip'],
```

Comment:
Why not adding a switch such as `compute_surface` to let the user control if the surface levels are extracted? At the moment, one would need to remove all `srf_var` from the config `WOA.yml` file.

Reply:
I thought this way the cmorizer could be more flexible in the case of future updates and I didn't liked the idea of hardcoding the 3D to 2D variable names correspondence in the script. 
Moreover, surface variable name lack of a certain standardization ... 
for example,  temperature is named in 3D `thetao` and in 2D `tos`.